pyccel/codegen/printing/ccode.py

```diff
@@ -702,13 +702,13 @@ def _new_slice_with_processed_arguments(_slice, array_size, allow_negative_index
         # negative start and end in slice
         if isinstance(start, PyccelUnarySub) and isinstance(start.args[0], LiteralInteger):
             start = PyccelMinus(array_size, start.args[0])
-        elif allow_negative_index and not isinstance(start, LiteralInteger):
+        elif allow_negative_index and not isinstance(start, (LiteralInteger, PyccelArraySize)):
```

Comment:
Why is this needed? What happens if I do:
```
@types('int[:]')
def f(a):
    import numpy as np
    s = np.shape(a)[0]
    b = np.zeros(s+1)
    print(b[s])
```
?

Reply:
that condition will occur only in handling a slice object and because we know that an array size will always be positive we don't need to have to add a ternary operator